src/main/starlark/builtins_bzl/common/cc/compile/compile.bzl

```
@@ -436,6 +493,602 @@ def _should_provide_header_modules(
+            ddi_file = source_to_ddi_file_map.get(source_artifact),
+            modules_info_file = modules_info_file,
+            modmap_file = modmap_file,
+            modmap_input_file = modmap_input_file,
+        )
+        cpp_compile_action_builder = cc_internal.create_cpp_compile_action_builder(
+            action_construction_context = action_construction_context,
+            cc_compilation_context = cc_compilation_context,
+            cc_toolchain = cc_toolchain,
+            configuration = configuration,
+            copts_filter = copts_filter,
+            feature_configuration = feature_configuration,
+            semantics = native_cc_semantics,
+            source_artifact = source_artifact,
+            additional_compilation_inputs = additional_compilation_inputs,
+            additional_include_scanning_roots = additional_include_scanning_roots,
+            module_files = all_module_files,
+            modmap_file = modmap_file,
+            modmap_input_file = modmap_input_file,
+            action_name = ACTION_NAMES.cpp20_module_compile,
+            additional_outputs = [module_file],
+        )
+
+        compiled_basenames.add(_basename_without_extension(source_artifact))
+
+        cc_internal.create_compile_source_action_from_builder(
+            action_construction_context = action_construction_context,
+            cc_compilation_context = cc_compilation_context,
+            cc_toolchain = cc_toolchain,
+            configuration = configuration,
+            conlyopts = conlyopts,
+            copts = copts,
+            cpp_configuration = cpp_configuration,
+            cxxopts = cxxopts,
+            fdo_context = fdo_context,
+            auxiliary_fdo_inputs = auxiliary_fdo_inputs,
+            feature_configuration = feature_configuration,
+            generate_no_pic_action = not use_pic,
+            generate_pic_action = use_pic,
+            label = label,
+            common_compile_build_variables = common_compile_build_variables,
+            fdo_build_variables = fdo_build_variables,
+            cpp_semantics = native_cc_semantics,
+            source_label = source_label,
+            output_name = output_name,
+            outputs = outputs,
+            source_artifact = source_artifact,
+            cpp_compile_action_builder = cpp_compile_action_builder,
+            output_category = artifact_category.CLIF_OUTPUT_PROTO if cpp_source.type == CPP_SOURCE_TYPE_CLIF_INPUT_PROTO else artifact_category.OBJECT_FILE,
+            cpp_module_map = cc_compilation_context.module_map(),
+            add_object = True,
+            enable_coverage = is_code_coverage_enabled,
+            generate_dwo = should_create_per_object_debug_info(feature_configuration, cpp_configuration),
+            bitcode_output = bitcode_output,
+            additional_build_variables = {
+                "cpp_module_output_file": module_file.path,
+                "cpp_module_modmap_file": modmap_file.path
```

```suggestion
                "cpp_module_output_file": module_file,
                "cpp_module_modmap_file": modmap_file,
```
should work just as well and support path mapping